Skip auth check in case of localhost usage

vkuznet opened the pull request:
Fixes #11461 

#### Status
ready

#### Description
Provide simple fix to check HTTP headers and skip authentication if localhost is used.

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs

#### External dependencies / deployment changes

vkuznet committed:
Skip auth check in case of localhost usage

amaltaro reviewed (changes requested):
It looks like unit tests need to be updated as well.

vkuznet committed:
Change unit test to allow localhost

vkuznet commented:
@amaltaro , I fixed `Simple_t` unit test by changing status, but I added appropriate comment. Please see in a code. The rest of failure in unit test are due to unstable tests.

amaltaro commented on src/python/WMCore/REST/Auth.py:
```diff
@@ -25,6 +25,9 @@ def user_info_from_headers(key, verbose=False):
     # Reject if request was not authenticated.
     if 'cms-auth-status' not in headers:
+        if headers.get('Remote-Addr', '') == '127.0.0.1':
+            # since we are using localhost access we can skip authentication
```
Valentin, I would suggest to take the same approach as the block below. So, please add a log here to identify that unauthenticated access has been performed (e.g.):
```
log("DEBUG: non-authenticated request performed")
```

vkuznet replied:
done

amaltaro reviewed (commented):
Valentin, I have made just a minor suggestion in the code. Thanks

amaltaro commented:
@mapellidario @belforte this might be handy as well when debugging CRABServer from localhost. Just FYI

belforte commented:
thanks Alan. But AFAICT we only had to worry about authentication in the old days when running CMSWEB FE and CRABServer on a VM. Now we do all work in K8s pods. Maybe I miss some other improvement which you guys did which makes testing CMSWEB apps even easier ?

amaltaro commented:
Valentin (and Todor) have been running specific WMCore services in their own node/laptop, with this change in, there is no need to have frontends performing the authentication part and adding extra HTTP request headers to the http call.

I haven't tried this myself though as I didn't try to run any of the central services in my own laptop :)

vkuznet commented:
Stefano, the changes I made will allow to run, debug code in local environment, eg laptop, without requiring FE presence.

Sent from Proton Mail mobile

-------- Original Message --------
On Feb 8, 2023, 5:56 PM, Stefano Belforte wrote:

> thanks Alan. But AFAICT we only had to worry about authentication in the old days when running CMSWEB FE and CRABServer on a VM. Now we do all work in K8s pods. Maybe I miss some other improvement which you guys did which makes testing CMSWEB apps even easier ?
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/dmwm/WMCore/pull/11462#issuecomment-1423346347), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AAA6RUW4HEBQPOIDWGGR6E3WWQQCPANCNFSM6AAAAAAUMNVUGY).
> You are receiving this because you were assigned.Message ID: ***@***.***>

belforte commented:
I see. Thanks Valentin and Alan !
